tools/releaseTools.psm1

```diff
@@ -723,6 +723,74 @@ function Get-PRBackportReport {
         $prs
     }
 }
+enum RemoteType {
+    GitHub
+    AzureRepo
+}
+
+function Get-UpstreamInfo {
+    param(
+        [Parameter(Mandatory=$true)]
+        [string[]]$Upstream,
+
+        [Parameter(Mandatory=$true)]
+        [string]$UpstreamRemote
+    )
+
+    $upstreamName = '(powershell(core)?)(/_git)?/(powershell)'
+    $pattern = "^$UpstreamRemote\s*(.*)\:(.*/([-\w.]+)/)?$upstreamName(\.git)?.*fetch"
+    Write-Verbose -Verbose "searching for an upstream with regex: '$pattern'"
+    $upstream = $upstream | Where-Object { $_ -match $pattern }
+
+    Write-Verbose -Verbose "found $upstream"
+
+    if (!$upstream) {
+        throw "Please create an upstream remote that points to $upstreamName"
+    }
+
+    $matches | Format-Table | Out-String -Stream -Width 9999 | Write-Verbose
```

Comment:
`$matches` will only contain the matching results from the last item of `$Upstream` that matches `$pattern`, but by looking at the code, there could be multiple items in `$Upstream` that match `$pattern`.